app.js

```diff
@@ -5,18 +5,44 @@ const {
   invalidEndpoint,
 } = require("./controllers/topics.controller");
 
+const {getArticleById} = require("./controllers/articles.controller")
+
 const {
   getEndpoints,
 } = require("./controllers/api.controller");
 
 app.get("/api", getEndpoints);
 
-app.get("/api/topics", getTopics)
+app.get("/api/topics", getTopics);
+
+app.get("/api/articles/:article_id", getArticleById);
 
 
 //should come after all other endpoints
 app.all("/*", invalidEndpoint)
 
+app.use((err, req, res, next) => {
+  if (err.msg && err.status) {
+    res.status(err.status).send({ msg: err.msg });
+  } else {
+    next(err);
+  }
+});
+
+app.use((err, req, res, next) => {
+  if (err.code === "22P02") {
+    res.status(400).send({ msg: "Bad request" });
+  } else {
+    next(err);
+  }
+});
+
```

Comment:
^^ at some point, your error handling is going to become a behemouth in this app.js file, it might be nice to split it off into an errors.controller.js file and make references to the functions here?